src/memory_dump/dump.c

```diff
@@ -358,213 +69,65 @@ void copy(char *source, char *dest) {
   fclose(output);
 }
 
-static void lock_mem(void) {
-
-#ifdef _DEBUG
-  printf("lock_mem()\n");
-#endif
-  char *p;
-  FILE *maps = fopen("/proc/self/maps", "r");
-
-  if (!maps)
-    errx(EXIT_FAILURE, "Error reading the memory using /proc/ interface");
-
-  char *addresses[65536];
-  char buf[BUFSIZ + 1];
-
-  size_t start, end;
-  int counter = 0;
-  char *start_of_stack, *end_of_stack;
-  off64_t errnol = (off64_t) round_to_page(__errno_location());
-
-  int count = 0;
-
-  while (fgets(buf, BUFSIZ, maps)) {
-    off64_t start, end;
-
-    //fprintf(stderr, "%s", buf);
-    sscanf(buf, "%lx-%lx", &start, &end);
-
-    /* Stack is special, it should be protected last
-       because we are using it */
-    if (strstr(buf, "stack") != NULL) {
-      end_of_stack = (char *)end;
-      start_of_stack = (char *)start;
-      continue;
-    }
-    /* Ignore libc pages */
-    if (strstr(buf, "linux-gnu") != NULL)
-      continue;
-    /* Ignore libc special mem zones */
-    if (strstr(buf, "r-xp") != NULL)
-      continue;
-    /* Ignore vsyscall special mem zones */
-    if (strstr(buf, "vsyscall") != NULL)
-      continue;
-
-    /* Ignore libdump mem zones */
-    if (strstr(buf, "libcere_dump.so") != NULL)
-      continue;
-
-    /* Ignore libdump vdso zones */
-    if (strstr(buf, "vdso") != NULL)
-      continue;
-
-    /* Ignore vvar zone (cf. https://lkml.org/lkml/2015/3/12/602)*/
-    if (strstr(buf, "vvar") != NULL)
-      continue;
-
-    /* Ignore alreay protected pages */
-    if (strstr(buf, "---p") != NULL)
-      continue;
-
-    char *page_start = round_to_page((char *)start);
-
-    /* syscall() and tracing SIGSEGV handler require access to errno.  We
-       must keep errno location unlocked to avoid trigering an infinite
-       segfault. */
-
-    if ((off64_t) page_start <= errnol && errnol < end) {
-      /* We split the range containing errnol into two segments */
+void dump_init(void) {
+  
+  state.mtrace_active = false;
 
-      /* If the left segment is non-empty, we must lock it */
-      if ((off64_t) page_start < errnol) {
-        assert(count < 65536);
-        addresses[count++] = (char *)page_start;
-        addresses[count++] = (char *)errnol;
-      }
+  char tracer_path[] = "/usr/local/bin/tracer";
```

Comment:
"/usr/local/bin/tracer" should not be hardcoded. But you can suppose that it is in the PATH.

Btw, `tracer` is way too generic, it should be renamed to `cere-tracer`